central/graphql/resolvers/node_vulnerabilities.go

```diff
@@ -80,8 +82,366 @@ func (resolver *Resolver) NodeVulnCounter(ctx context.Context, args RawQuery) (*
+	if err := readCVEs(ctx); err != nil {
+		return nil, err
+	}
+	vuln, exists, err := resolver.NodeCVEDataStore.Get(ctx, string(*args.ID))
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errors.Errorf("node cve not found: %s", string(*args.ID))
+	}
+	vulnResolver, err := resolver.wrapNodeCVE(vuln, true, nil)
+	if err != nil {
+		return nil, err
+	}
+	vulnResolver.ctx = ctx
+	return vulnResolver, nil
+}
+
+func (resolver *Resolver) nodeCvesV2(ctx context.Context, args PaginatedQuery) ([]NodeVulnerabilityResolver, error) {
+	if err := readCVEs(ctx); err != nil {
+		return nil, err
+	}
+	query, err := args.AsV1QueryOrEmpty()
+	if err != nil {
+		return nil, err
+	}
+
+	vulnResolvers, err := resolver.nodeCvesV2Query(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := make([]NodeVulnerabilityResolver, 0, len(vulnResolvers))
+	for _, res := range vulnResolvers {
+		res.ctx = ctx
+		ret = append(ret, res)
+	}
+	return ret, nil
+}
+
+func (resolver *Resolver) nodeCvesV2Query(ctx context.Context, query *v1.Query) ([]*nodeCVEResolver, error) {
+	vulnLoader, err := loaders.GetNodeCVELoader(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	query = tryUnsuppressedQuery(query)
+
+	vulns, err := vulnLoader.FromQuery(ctx, query)
+	vulnResolvers, err := resolver.wrapNodeCVEs(vulns, err)
+	return vulnResolvers, err
+}
+
+func (resolver *Resolver) nodeCveCountV2(ctx context.Context, args RawQuery) (int32, error) {
+	if err := readCVEs(ctx); err != nil {
+		return 0, err
+	}
+	query, err := args.AsV1QueryOrEmpty()
+	if err != nil {
+		return 0, err
+	}
+
+	vulnLoader, err := loaders.GetNodeCVELoader(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	query = tryUnsuppressedQuery(query)
+	return vulnLoader.CountFromQuery(ctx, query)
+}
+
+func (resolver *Resolver) nodeCveCounterV2(ctx context.Context, args RawQuery) (*VulnerabilityCounterResolver, error) {
+	if err := readCVEs(ctx); err != nil {
+		return nil, err
+	}
+	query, err := args.AsV1QueryOrEmpty()
+	if err != nil {
+		return nil, err
+	}
+	return resolver.nodeCveCounterV2Query(ctx, query)
+}
+
+func (resolver *Resolver) nodeCveCounterV2Query(ctx context.Context, query *v1.Query) (*VulnerabilityCounterResolver, error) {
+	vulnLoader, err := loaders.GetNodeCVELoader(ctx)
+	if err != nil {
+		return nil, err
+	}
+	query = tryUnsuppressedQuery(query)
+	fixableVulnsQuery := search.ConjunctionQuery(query, search.NewQueryBuilder().AddBools(search.Fixable, true).ProtoQuery())
+	fixableVulns, err := vulnLoader.FromQuery(ctx, fixableVulnsQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	unFixableVulnsQuery := search.ConjunctionQuery(query, search.NewQueryBuilder().AddBools(search.Fixable, false).ProtoQuery())
+	unFixableCVEs, err := vulnLoader.FromQuery(ctx, unFixableVulnsQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	return mapNodeCVEsToVulnerabilityCounter(fixableVulns, unFixableCVEs), nil
+}
```

Comment:
It feels like the logic of this function could be collapsed into logic higher up, similar to previous comment